lib/microformats/format_parser.rb

```diff
@@ -347,10 +122,262 @@ def check_for_h_properties
+      else
+        # else if .h-x>img[src]:only-of-type:not[.h-*] then use that img src for photo
+        child_img_tags_with_src = element.children.select do |child|
+          child.is_a?(Nokogiri::XML::Element) && child.name == 'img' && !child.attribute('src').nil?
+        end
+
+        if child_img_tags_with_src.count == 1
+          node = child_img_tags_with_src.first
+
+          @properties['photo'] = [node.attribute('src').value.strip] if format_classes(node).empty?
+        end
+
+        if @properties['photo'].nil?
+          # else if .h-x>object[data]:only-of-type:not[.h-*] then use that object's data for photo
+          child_object_tags_with_data = element.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'object' && !child.attribute('data').nil?
+          end
+
+          if child_object_tags_with_data.count == 1
+            node = child_object_tags_with_data.first
+
+            if format_classes(node).empty?
+              @properties['photo'] = [node.attribute('data').value.strip]
+            end
+          end
+        end
+
+        child_elements = element.children.reject { |child| child.is_a?(Nokogiri::XML::Text) }
+
+        if @properties['photo'].nil? && child_elements.count == 1 && format_classes(child_elements.first).empty?
+          # else if .h-x>:only-child:not[.h-*]>img[src]:only-of-type:not[.h-*], then use that img's src for photo
+          child_img_tags_with_src = child_elements.first.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'img' && !child.attribute('src').nil?
+          end
+
+          if child_img_tags_with_src.count == 1
+            node = child_img_tags_with_src.first
+
+            if format_classes(node).empty?
+              @properties['photo'] = [node.attribute('src').value.strip]
+            end
+          end
+
+          if @properties['photo'].nil?
+            # else if .h-x>:only-child:not[.h-*]>object[data]:only-of-type:not[.h-*], then use that object's data for photo
+            child_object_tags_with_data = child_elements.first.children.select do |child|
+              child.is_a?(Nokogiri::XML::Element) && child.name == 'object' && !child.attribute('data').nil?
+            end
+
+            if child_object_tags_with_data.count == 1
+              node = child_object_tags_with_data.first
+
+              if format_classes(node).empty?
+                @properties['photo'] = [node.attribute('data').value.strip]
+              end
+            end
+          end
+        end
+      end
+
+      unless @properties['photo'].nil?
+        @properties['photo'] = [Microformats::AbsoluteUri.new(@properties['photo'].first, base: @base).absolutize]
+      end
+    end
+
+    def imply_url(element)
+      unless @properties['url'].nil? && !@found_prefixes.include?(:u) && !@found_prefixes.include?(:h) && @children.empty?
+        return nil
+      end
```

Comment:
Similar to above, this can be shortened up to:

```ruby
return unless @properties['url'].nil? && !@found_prefixes.include?(:u) && !@found_prefixes.include?(:h) && @children.empty?
```